SpatialGDKEditor/Source/Public/SpatialGDKEditorSettings.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+#pragma once
+
+#include "SpatialGDKEditorSettings.generated.h"
+
+UCLASS(config = EditorPerProjectUserSettings, defaultconfig)
+class SPATIALGDKEDITOR_API USpatialGDKEditorSettings : public UObject
+{
+	GENERATED_BODY()
+
+public:
+	USpatialGDKEditorSettings(const FObjectInitializer& ObjectInitializer);
+
+private:
+	/** Path to the directory containing the SpatialOS-related files. */
+	UPROPERTY(EditAnywhere, config, Category = "Configuration", meta = (ConfigRestartRequired = false, DisplayName = "SpatialOS directory"))
+		FDirectoryPath SpatialOSDirectory;
+
+public:
+	/** Generate schema for all classes supported by the GDK */
+	UPROPERTY(EditAnywhere, config, Category = "Schema Generation", meta = (ConfigRestartRequired = false, DisplayName = "Generate schema for all supported classes"))
+	bool bGenerateSchemaForAllSupportedClasses;
+
+private:
+	/** Path to your SpatialOS snapshot. */
+	UPROPERTY(EditAnywhere, config, Category = "Configuration", meta = (ConfigRestartRequired = false, DisplayName = "Snapshot path"))
+		FDirectoryPath SpatialOSSnapshotPath;
+
+	/** Generated schema output path */
+	UPROPERTY(EditAnywhere, config, Category = "Schema Generation", meta = (ConfigRestartRequired = false, DisplayName = "Output path for the generated schemas"))
+		FDirectoryPath GeneratedSchemaOutputFolder;
+
+public:
+	UFUNCTION()
```

Comment:
Do any of these need to be `UFUNCTION`s? (I know they were `UFUNCTION`s even before this PR, but it doesn't seem like that was necessary)

Reply:
I honestly have no clue. I don't fully know what `UFUNCTION`s are even for, so I just made sure the existing functionality was preserved. This sounds like a separate issue from this PR tho.

Reply:
`UFUNCTION`s are fancy functions that allow Unreal to do special things with them (e.g. calling from blueprints, using them in delegates by name, etc.) More info here: https://docs.unrealengine.com/en-us/Programming/UnrealArchitecture/Reference/Functions

I think in this case, if these C++ functions don't fall into one of the use cases of `UFUNCTION`s, we should remove the `UFUNCTION` marco just to clean up the file.

Reply:
Is this a question that the UnrealGDK team itself is better suited to answer? For the most part, I only migrated the code, so I feel I have an inadequate understanding of the intended usages of the original code as written.
I'd be more than happy to make the changes as a part of this PR, but I'm probably not the best to make the decision on whether it should be done or not. This is you guys' code :)